pig/src/main/java/org/jboss/pnc/bacon/pig/impl/nvr/NvrListGenerator.java

```diff
@@ -38,31 +37,31 @@
     private NvrListGenerator() {
     }
 
-    public static boolean generateNvrList(Map<String, Collection<String>> checksums, String targetPath) {
+    public static boolean generateNvrList(Map<String, Collection<String>> checksums, Path targetPath) {
         log.info("Generating NVR list for {} checksums and saving result to {}", checksums.size(), targetPath);
 
         List<KojiBuild> builds = BuildFinderUtils.findBuilds(checksums, true);
-        File outputDirectory = new File(FilenameUtils.getPath(targetPath));
+        File outputDirectory = targetPath.normalize().getParent().toFile();
         Report nvrReport = new NVRReport(outputDirectory, builds);
-        String basename = nvrReport.getBaseFilename() + ".txt";
+        nvrReport.setBaseFilename(baseFileName(targetPath));
 
         try {
             nvrReport.outputText();
         } catch (IOException e) {
-            log.error("Failed to write file {} to {}: {}", basename, outputDirectory, e.getMessage(), e);
+            log.error("Failed to write file {} to {}: {}", targetPath, outputDirectory, e.getMessage(), e);
             return false;
         }
 
-        File srcFile = new File(outputDirectory, basename);
-        File destFile = new File(targetPath);
+        return true;
+    }
 
-        try {
-            FileUtils.copyFile(srcFile, destFile, true);
-        } catch (IOException e) {
-            log.error("Failed to copy file {} to {}: {}", srcFile, destFile, e.getMessage(), e);
-            return false;
+    private static String baseFileName(Path targetPath) {
```

Comment:
There is also https://commons.apache.org/proper/commons-io/javadocs/api-release/org/apache/commons/io/FilenameUtils.html#getBaseName-java.lang.String- whichamke make your code smaller.